gallery/src/components/NxModal/NxModalFormErrorExample.tsx

```diff
@@ -6,12 +6,23 @@
  */
 import React, {useState} from 'react';
 
-import {NxModal, NxFontAwesomeIcon} from '@sonatype/react-shared-components';
+import {NxModal, NxFontAwesomeIcon, NxTextInput, NxLoadError} from '@sonatype/react-shared-components';
 import {faAngry} from '@fortawesome/free-solid-svg-icons';
+import { initialState, userInput } from '@sonatype/react-shared-components/components/NxTextInput/stateHelpers';
 
 export default function NxModalFormErrorExample() {
   const [showModal, setShowModal] = useState(false);
   const modalCloseHandler = () => setShowModal(false);
+  const [state, setState] = useState(initialState(''));
+  const [error, setError] = useState<string | null>('');
+
+  function retryHandler() {
+    // lets say the retried action succeeded this time
+    setError(null);
+  }
```

Comment:
That `onClick` handler isn't the right way to do this, and if you notice, clicking _anywhere_ on the alert will close the modal at the moment.  Remove that `onClick` handler and try this:

```suggestion
  function retryHandler() {
    // lets say the retried action succeeded this time
    setShowModal(false);
  }
```